internal/controllers/machineset/machineset_controller.go

```diff
@@ -529,45 +539,53 @@ func (r *Reconciler) syncReplicas(ctx context.Context, ms *clusterv1.MachineSet,
 	return nil
 }
 
-// getNewMachine creates a new Machine object. The name of the newly created resource is going
-// to be created by the API server, we set the generateName field.
-func (r *Reconciler) getNewMachine(machineSet *clusterv1.MachineSet) *clusterv1.Machine {
-	gv := clusterv1.GroupVersion
-	machine := &clusterv1.Machine{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: fmt.Sprintf("%s-", machineSet.Name),
-			// Note: by setting the ownerRef on creation we signal to the Machine controller that this is not a stand-alone Machine.
-			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(machineSet, machineSetKind)},
-			Namespace:       machineSet.Namespace,
-			Labels:          make(map[string]string),
-			Annotations:     machineSet.Spec.Template.Annotations,
-		},
-		TypeMeta: metav1.TypeMeta{
-			Kind:       gv.WithKind("Machine").Kind,
-			APIVersion: gv.String(),
-		},
-		Spec: machineSet.Spec.Template.Spec,
-	}
-	machine.Spec.ClusterName = machineSet.Spec.ClusterName
-
-	// Set the labels from machineSet.Spec.Template.Labels as labels for the new Machine.
-	// Note: We can't just set `machineSet.Spec.Template.Labels` directly and thus "share" the labels
-	// map between Machine and machineSet.Spec.Template.Labels. This would mean that adding the
-	// MachineSetNameLabel and MachineDeploymentNameLabel later on the Machine would also add the labels
-	// to machineSet.Spec.Template.Labels and thus modify the labels of the MachineSet.
+func (r *Reconciler) computeDesiredMachine(machineSet *clusterv1.MachineSet, existingMachine *clusterv1.Machine) *clusterv1.Machine {
+	desiredMachine := &clusterv1.Machine{}
+
+	desiredMachine.SetGroupVersionKind(clusterv1.GroupVersion.WithKind("Machine"))
```

Comment:
Q: Would it be easier to read if we just define the first few properties as a struct like we did before?
(~ l.545-553 + set an empty map for labels and annotations)